scripts/lib/CIME/namelist.py

```diff
@@ -884,25 +975,54 @@ def get_value(self, variable_name):
         else:
             return [u'']
 
-    def set_variable_value(self, group_name, variable_name, value):
+    def set_variable_value(self, group_name, variable_name, value, var_size=1):
         """Set the value of the specified variable.
 
         >>> x = parse(text='&foo bar=1 /')
+        >>> x.get_variable_value('foo', 'bar')
+        [u'1']
+        >>> x.set_variable_value('foo', 'bar(2)', [u'3'], var_size=4)
+        >>> x.get_variable_value('foo', 'bar')
+        [u'1', u'3']
         >>> x.set_variable_value('foo', 'bar', [u'2'])
+        >>> x.get_variable_value('foo', 'bar')
+        [u'2', u'3']
         >>> x.set_variable_value('foo', 'bazz', [u'3'])
         >>> x.set_variable_value('Brack', 'baR', [u'4'])
-        >>> x.get_variable_value('foo', 'bar')
-        [u'2']
         >>> x.get_variable_value('foo', 'bazz')
         [u'3']
         >>> x.get_variable_value('brack', 'bar')
         [u'4']
+        >>> x.set_variable_value('foo', 'red(2:6:2)', [u'2', u'4', u'6'], var_size=12)
+        >>> x.get_variable_value('foo', 'red')
+        ['', u'2', '', u'4', '', u'6']
         """
         group_name = group_name.lower()
-        variable_name = variable_name.lower()
+
+        minindex, maxindex, step = get_fortran_variable_indices(variable_name, var_size)
+        variable_name = get_fortran_name_only(variable_name.lower())
+
+        expect(minindex > 0, "Indices < 1 not supported in CIME interface to fortran namelists... lower bound=%s"%minindex)
+
         if group_name not in self._groups:
             self._groups[group_name] = {}
-        self._groups[group_name][variable_name] = value
+        tlen = 1
+        if variable_name in self._groups[group_name]:
+            tlen = len(self._groups[group_name][variable_name])
+        else:
+            tlen = 1
+            self._groups[group_name][variable_name] = ['']
+
+        if minindex > tlen:
+            self._groups[group_name][variable_name].extend(['']*(minindex-tlen-1))
+
+        for i in range(minindex-1, maxindex+step, step):
+            while len(self._groups[group_name][variable_name]) < i+1:
```

Comment:
I think this logic is incorrect. For instance:
`dtlimit(20:22:2) = 20.0, 22.0, 24.0`
should generate an error since the slice only indicates two elements but you accept (and set) three. This is an ERROR in Fortran and should be trapped. If you trap the error earlier, then you can maybe keep the logic.
I believe the correct number of steps is:
`1 + ((maxindex - minindex) / step)`